images/singleuser/Dockerfile

```diff
@@ -199,4 +204,8 @@ COPY --chown=tools.paws:tools.paws hide_clusters_tab.css /home/paws/.jupyter/cus
 # Setup custom config as needed
 COPY jupyter_notebook_config.py /srv/paws/etc/jupyter/jupyter_notebook_config.py
 
+# install julia kernel
+RUN julia -e 'using Pkg; Pkg.add("IJulia")'
+RUN bash -c "cp -r /home/paws/.local/share/jupyter/kernels/julia* /srv/paws/share/jupyter/kernels/"
```

Comment:
I could use some opinion on how the kernel.json file gets put in place. Not quite clear on how the others do it, they don't use a cp, though perhaps it is because they are more pip based that they know where to put the file as part of their install?

Reply:
Check out https://github.com/berkeley-dsep-infra/datahub/blob/d55906952b0f0819da2c85d752a9ac4a593060a1/deployments/julia/image/Dockerfile#L62 - I think that's the best way to install the kernel. You'll need to figure out JUYPTER_DATA_DIR - I think if you run `jupyter --path` in the container, it'll tell you where it looks for kernels. You should put it in the path that's not under /home/paws - $HOME is always mounted over by our NFS, so anything under $HOME in the docker image won't be available once the user starts.